### PR TITLE
Add use import statements for PHP Snippets

### PR DESCRIPTION
**Pull request by shemogumbe**
## Overview

Use statements added at the top of php snippets for model and request builder namespaces.

### Demo

```
<?php
use Microsoft\Graph\Graph;
use Microsoft\Graph\Generated\Users\Item\Authentication\Operations\Item;


$graphServiceClient = new GraphServiceClient($tokenRequestContext, $scopes);

$requestConfiguration = new LongRunningOperationItemRequestBuilderGetRequestConfiguration();
$queryParameters = LongRunningOperationItemRequestBuilderGetRequestConfiguration::createQueryParameters();
$queryParameters->aadgdc = "AM4P";
$queryParameters->aadgsu = "ssprprod-a";
$requestConfiguration->queryParameters = $queryParameters;


$result = $graphServiceClient->users()->byUserId('user-id')->authentication()->operations()->byLongRunningOperationId('longRunningOperation-id')->get($requestConfiguration)->wait();

```

## Testing Instructions

Check this branch https://github.com/microsoftgraph/microsoft-graph-docs/blob/preview-snippet-generation/140119/includes/snippets/php/v1/delta-query-groups-initial-request-php-snippets.md

### CODE REVIEW

**Comment by timayabi2020** [comment]
There are a couple of sona source warnings. Maybe you should have a look.

**Review by SilasKenneth — CHANGES_REQUESTED** [review]

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/PhpGenerator.cs [hunk 1]
```diff
@@ -106,9 +106,51 @@ public string GenerateCodeSnippet(SnippetModel snippetModel)
...
+            foreach (var import in imports)
+            {
+                switch (import.Kind)
+                {
+                    case ImportKind.Model:
+                        snippetImports.Add($"{modelImportPrefix};");
+                        // check if model has a nested namespace and append it to the import statement
+                        break;
+                    case ImportKind.Path:
+                        if (import.Path != null && import.RequestBuilderName != null)
```
Comment (andrueastman) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/PhpGenerator.cs:142]:
```suggestion
                        if (!string.IsNullOrEmpty(import.Path) && !string.IsNullOrEmpty(import.RequestBuilderName))
```
We are checking a string.

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/PhpGenerator.cs [hunk 2]
```diff
@@ -106,9 +106,51 @@ public string GenerateCodeSnippet(SnippetModel snippetModel)
...
+                        {
+                            //construct path to request builder
+                            var importPath = import.Path.Split('.')
+                                .Select(s => s.ToFirstCharacterUpperCase()).ToArray();
```
Comment (andrueastman) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/PhpGenerator.cs:146]:
```suggestion
                                .Select(static s => s.ToFirstCharacterUpperCase()).ToArray();
```

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/PhpGenerator.cs [hunk 3]
```diff
@@ -106,9 +106,51 @@ public string GenerateCodeSnippet(SnippetModel snippetModel)
...
+    private static HashSet<string> GetImportStatements(SnippetModel snippetModel)
+        {
+            const string modelImportPrefix = "use Microsoft\\Graph\\Generated\\Models";
+            const string requestBuilderImportPrefix = "use Microsoft\\Graph\\Generated";
+
+            var snippetImports = new HashSet<string>();
 
+            snippetImports.Add("use Microsoft\\Graph\\Graph;");
+
+            var importsGenerator = new ImportsGenerator();
```
Comment (andrueastman) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/PhpGenerator.cs:131]: The `ImportsGenerator` could probably be a class variable. There is no need to create an instance every time the function is called.  Alternatively, the `GenerateImportTemplates` and `ImportsGenerator` and can be a static function and class so we don't have to create an instance when we need it as `GenerateImportTemplates` is a pure function.

File: CodeSnippetsReflection.OpenAPI.Test/PhpImportGeneratorTests.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            ]
+        }";
+        using var requestPayload = new HttpRequestMessage(HttpMethod.Post, $"{ServiceRootUrl}/me/messages/")
+        {
+            Content = new StringContent(bodyContent, Encoding.UTF8, "application/json")
+        };
+        var snippetModel = new SnippetModel(requestPayload, ServiceRootUrl, await GetV1SnippetMetadata());
+        var result = _generator.GenerateCodeSnippet(snippetModel);
+        Assert.Contains("use Microsoft\\Graph\\Graph;", result);
+        Assert.Contains("use Microsoft\\Graph\\Generated\\Models;", result);
```
[the remark this answers is not in the document]
Comment (shemogumbe) [reply CodeSnippetsReflection.OpenAPI.Test/PhpImportGeneratorTests.cs:71]: Makes sense as a way of making the code more explicit and improving performance, that's how it's done in Python.

**Review by SilasKenneth — APPROVED** [review]

**Pull request merged**